cmd/main.go

```diff
@@ -1,426 +1,46 @@
 package main
 
 import (
-	"bytes"
-	"crypto/sha256"
-	"encoding/base64"
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"log"
 	"net/http"
-	"os"
-	"strings"
-	"sync"
 	"time"
 
-	"github.com/blevesearch/bleve"
-	"github.com/gorilla/mux"
-	"github.com/jinzhu/copier"
-	"github.com/opds-community/libopds2-go/opds2"
-	"github.com/readium/r2-streamer-go/pkg/decoder/lcp"
-	"github.com/readium/r2-streamer-go/pkg/fetcher"
-	"github.com/readium/r2-streamer-go/pkg/parser"
-	"github.com/readium/r2-streamer-go/pkg/pub"
-	"github.com/readium/r2-streamer-go/pkg/searcher"
-	"github.com/urfave/negroni"
+	streamer "github.com/readium/r2-streamer-go"
+	"github.com/readium/r2-streamer-go/internal/config"
+	"github.com/readium/r2-streamer-go/internal/logging"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 )
 
-type currentBook struct {
-	filename    string
-	publication pub.Publication
-	timestamp   time.Time
-	bleveIndex  bleve.Index
-	indexed     bool
+func init() {
```

Comment:
We will probably add other CLI tools later on in this repo. So in the spirit of the monorepo, I suggest we add a new package `cmd/server/` or `cmd/pubserver` to contain all the stuff related to the "plug and play" publication server, including:

* A `README.md` dedicated to the publication server.
* The sample `configs/` folder and the `internal/config` package.
* `opds.go`, `server.go`, `structs.go`
    * Maybe some of the code inside could later on (not important now) be extracted in a public `pkg/` package to be used as building blocks for custom servers.

Reply:
@mickael-menu is the structure in the latest commit what you are imagining? I don't really like having the cmd/server full of a lot of files, having them in the main package means you can't for example use `go run cmd/server/main.go` anymore.

Reply:
I would keep under `cmd/server/` only the `main()` function and its direct dependencies (initialization of the dependency graph for example) and the rest can go either:
* in `cmd/server/internal` if it's only relevant to the plug and play server.
* in a public `pkg/` package if it's a useful building block that can be reused in custom server implementations.

But as long as we have a clear separation between each command and the public packages, feel free to structure as you prefer.